src/pages/Wallet/SendPage.tsx

```diff
@@ -97,32 +101,83 @@ const SendPage = () => {
+          addPendingTx({
+            txId: txSendResp.data.txId,
+            toAddress: address,
+            timestamp: new Date().getTime(),
+            amount: fullAmount,
+            type: 'transfer'
+          })
+        }
 
-        const signature = client.clique.transactionSign(txId, wallet.privateKey)
+        setSnackbarMessage({ text: 'Transaction sent!', type: 'success' })
+        history.push('/wallet')
+      } catch (e) {
+        // TODO: When API error codes are available, replace this substring check with a proper error code check
+        if (isHTTPError(e) && e.error?.detail && e.error.detail.includes('consolidating')) {
+          setIsConsolidateUTXOsModalVisible(true)
+        } else {
+          setSnackbarMessage({
+            text: getHumanReadableError(e, 'Error while sending the transaction'),
+            type: 'alert',
+            duration: 5000
+          })
+        }
+      }
 
-        const txSendResp = await client.clique.transactionSend(wallet.address, unsignedTx, signature)
+      setIsSending(false)
+    }
+  }
 
-        addPendingTx({
-          txId: txSendResp.data.txId,
-          toAddress: address,
-          timestamp: new Date().getTime(),
-          amount: fullAmount
-        })
+  const consolidateUTXOs = async () => {
+    if (client && wallet) {
+      setIsConsolidating(true)
+      try {
+        const txCreateResp = await client.clique.transactionConsolidateUTXOs(
+          wallet.publicKey,
+          wallet.address,
+          wallet.address
+        )
+        for (const { txId, unsignedTx } of txCreateResp.data.unsignedTxs) {
+          const txSendResp = await signAndSendTransaction(txId, unsignedTx, wallet.address, wallet.privateKey)
+          if (txSendResp) {
+            addPendingTx({
+              txId: txSendResp.data.txId,
+              toAddress: wallet.address,
+              timestamp: new Date().getTime(),
+              amount: '',
+              type: 'consolidation'
+            })
+          }
+        }
 
-        setSnackbarMessage({ text: 'Transaction sent!', type: 'success' })
-        history.push('/wallet')
+        setSnackbarMessage({
+          text: 'Consolidation process in progress. You can try to send your transaction again once the pending consolidation transactions have been confirmed.',
+          type: 'info',
+          duration: 15000
+        })
+        setIsConsolidateUTXOsModalVisible(false)
+        initialOnModalClose.current()
       } catch (e) {
         setSnackbarMessage({
-          text: getHumanReadableError(e, 'Error while sending the transaction'),
+          text: getHumanReadableError(e, 'Error while consolidating UTXOs'),
           type: 'alert',
           duration: 5000
         })
       }
-
-      setIsSending(false)
+      setIsConsolidating(false)
     }
   }
 
+  const signAndSendTransaction = async (txId: string, unsignedTx: string, toAddress: string, privateKey: string) => {
+    if (!client) return
```

Comment:
Should we throw an error if `client` is not defined? Or we could assume that `client` is define when this function is called

Reply:
We can assume it's defined, because without a `client`, the user wouldn't be able to login to the wallet.